src/official_interface.py

```diff
@@ -0,0 +1,82 @@
+"""競技システムインタフェース.
+
+競技システムとの通信を行うクラス.
+@author: nishijima515
+"""
+import requests
+import os
+# from image_processing import ImageProcessing
+from PIL import Image
+
+
+class ResponseError(Exception):
+    """レスポンスエラー用の例外."""
+
+    def __init__(self, message: str):
+        """コンストラクタ.
+
+        Args:
+            message (string): エラーメッセージ
+        """
+        super().__init__(message)
+
+
+class OfficialInterface:
+    """競技システムとの通信を行うクラス."""
+
+    SERVER_IP = "192.168.100.1"    # 競技システムのIPアドレス
+    TEAM_ID = 117                   # チームID
+
+    @classmethod
+    def upload_snap(cls, img_path: str) -> bool:
+        """指定された画像をアップロードする.
+
+        Args:
+            img_path (str): アップロードする画像のパス
+
+        Returns:
+            success (bool): 通信が成功したか(成功:true/失敗:false)
+        """
+        url = f"http://{cls.SERVER_IP}/snap"
+        # リクエストヘッダー
+        headers = {
+            "Content-Type": "image/jpeg"
+        }
+        # リクエストパラメータ
+        params = {
+            "id": cls.TEAM_ID
+        }
+        try:
+            # サイズが正しくない場合はリサイズする
+            img = Image.open(img_path)
+            width, height = img.size
+            if not (width == 800 and height == 600):
+                img = img.resize((800, 600))
+                img.save(img_path)
```

Comment:
一応、リサイズ後の画像を明示的にJPEGで保存するようにしたほうがいいかも
```suggestion
                img.save(img_path, format="JPEG")
```